tests/bluetooth/bsim/host/adv/encrypted/src/central.c

```diff
@@ -0,0 +1,96 @@
+/* Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "common.h"
+
+#define EXPECTED_DATA_SIZE 6
+static uint8_t expected_data[] = {0x5A, 0x65, 0x70, 0x68, 0x79, 0x72};
+BUILD_ASSERT(sizeof(expected_data) == EXPECTED_DATA_SIZE);
+
+static uint8_t session_key[] = {0xC0, 0xC1, 0xC2, 0xC3, 0xC4, 0xC5, 0xC6, 0xC7,
+				0xC8, 0xC9, 0xCA, 0xCB, 0xCC, 0xCD, 0xCE, 0xCF};
+BUILD_ASSERT(sizeof(session_key) == BT_EAD_KEY_SIZE);
+static uint8_t iv[] = {0xFB, 0x56, 0xE1, 0xDA, 0xDC, 0x7E, 0xAD, 0xF5};
+BUILD_ASSERT(sizeof(iv) == BT_EAD_IV_SIZE);
+
+static bt_addr_le_t prev_addr;
+
+static bool data_parse_cb(struct bt_data *data, void *user_data)
+{
+	if (data->type == BT_DATA_ENCRYPTED_AD_DATA) {
+		int err;
+		uint8_t encrypted_data_size = data->data_len;
+		uint8_t decrypted_data_size =
+			((encrypted_data_size - BT_EAD_RANDOMIZER_SIZE) - BT_EAD_AAD_SIZE) -
+			BT_EAD_MIC_SIZE;
+		uint8_t *decrypted_data = k_malloc(sizeof(decrypted_data[0]) * decrypted_data_size);
+
+		err = bt_ead_decrypt(session_key, iv, data->data, encrypted_data_size,
+				     decrypted_data);
+
+		if (err < 0) {
+			FAIL("Error during decryption\n");
+		}
+
+		if (memcmp(decrypted_data, expected_data, decrypted_data_size)) {
+			LOG_HEXDUMP_DBG(data, decrypted_data_size, "Decrypted data: ");
+			LOG_HEXDUMP_DBG(expected_data, decrypted_data_size, "Expected data: ");
+			FAIL("Decrypted data do not match expected data\n");
+		} else {
+			LOG_DBG("Data successfully decrypted");
+		}
+
+		k_free(decrypted_data);
+	}
+
+	return true;
+}
+
+static void device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
+			 struct net_buf_simple *ad)
+{
+	char addr_str[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(addr, addr_str, sizeof(addr_str));
+
+	LOG_DBG("Device found: %s (RSSI %d)", addr_str, rssi);
+
+	if (!bt_addr_le_eq(addr, &prev_addr)) {
+		bt_data_parse(ad, &data_parse_cb, NULL);
+
+		bt_addr_le_copy(&prev_addr, addr);
+	}
+}
+
+static void start_scan(void)
+{
+	int err;
+
+	err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, device_found);
+
+	if (err) {
+		FAIL("Scanning failed to start (err %d)\n", err);
+	}
+
+	LOG_DBG("Scanning successfully started");
+}
+
+void test_central(void)
+{
+	int err;
+
+	LOG_DBG("Central device");
+
+	bst_result = Passed;
```

Comment:
Aren't you setting the test to pass before even enabling the BT Stack here?

Reply:
Yes, this is intended. The test run for the whole simulation length. So by default, I set the test result to `Passed`. The test will fail only if it encounter a `FAIL(...)`.

Reply:
Not sure I understand. 

Why not set the `bst_result` to `In_progress`, and when you get to the point where you want to pass the test, set it to `Passed`, like all the other BabbleSim tests we have? 

Why do you want to run it until the simulation length?